sklearn/preprocessing/_target_encoder.py

```diff
@@ -0,0 +1,337 @@
+          feature, and should be sorted in case of numeric values.
+
+        The used categories is stored in the `categories_` fitted attribute.
+
+    target_type : {"auto", "continuous", "binary"}, default="auto"
+        Type of target.
+
+        - `"auto"` : Type of target is inferred with
+          :func:`~sklearn.utils.multiclass.type_of_target`.
+        - `"continuous"` : Continuous target
+        - `"binary"` : Binary target
+
+        .. note::
+            The type of target inferred with `"auto"` may not be the desired target
+            type used for modeling. For example, if the target consistent of integers
+            between 0 and 100, then :func:`~sklearn.utils.multiclass.type_of_target`
+            will infer the target as `"multiclass"`. In this case, setting
+            `target_type="continuous"` will the target as a regression problem. The
+            `target_type_` attribute gives the target type used by the encoder.
+
+    smooth : "auto" or float, default="auto"
+        The amount of mixing of the categorical encoding with the global target mean. A
+        larger `smooth` value will put more weight on the global target mean.
+        If `"auto"`, then `smooth` is set to an empirical Bayes estimate.
+
+    cv : int, default=5
+        Determines the number of folds in the cross-validation strategy used in
+        :meth:`fit_transform`. For classification targets, `StratifiedKFold` is used
+        and for continuous targets, `KFold` is used.
+
+    shuffle : bool, default=True
+        Whether to shuffle the data in :meth:`fit_transform` before splitting into
+        batches. Note that the samples within each split will not be shuffled.
+
+    random_state : int, RandomState instance or None, default=None
+        When `shuffle` is True, `random_state` affects the ordering of the
+        indices, which controls the randomness of each fold. Otherwise, this
+        parameter has no effect.
+        Pass an int for reproducible output across multiple function calls.
+        See :term:`Glossary <random_state>`.
+
+    Attributes
+    ----------
+    encodings_ : list of shape (n_features,) of ndarray
+        For feature `i`, `encodings_[i]` is the encoding matching the
+        categories listed in `categories_[i]`.
+
+    categories_ : list of shape (n_features,) of ndarray
+        The categories of each feature determined during fitting
+        (in order of the features in `X` and corresponding with the output
+        of :meth:`transform`).
+
+    target_type_ : str
+        Type of target.
+
+    encoding_mean_ : float
+        The overall mean of the target. This value is only used in :meth:`transform`
+        to encode categories.
+
+    n_features_in_ : int
+        Number of features seen during :term:`fit`.
+
+    feature_names_in_ : ndarray of shape (`n_features_in_`,)
+        Names of features seen during :term:`fit`. Defined only when `X`
+        has feature names that are all strings.
```

Comment:
Is this redundant with the categories_ attribute? Seems to me that the output will be identical.

Reply:
They give different information. `feature_names_in_` are the column names of the input dataframe and `categories_` are the actual categories.  For example:

```python
from sklearn.preprocessing import TargetEncoder
import pandas as pd

X_df = pd.DataFrame({"pet": ["dog", "cat", "snake"]})
y = [0, 1, 0]
enc = TargetEncoder().fit(X_df, y)

enc.feature_names_in_
# array(['pet'], dtype=object)

enc.categories_
# [array(['cat', 'dog', 'snake'], dtype=object)]
```

Reply:
Ok, I see, thanks, I was mixing features and categories.